rama-http/src/headers/x_robots_tag/element.rs

```diff
@@ -0,0 +1,46 @@
+use crate::headers::x_robots_tag::rule::Rule;
+use rama_core::error::OpaqueError;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Element {
```

Comment:
Depending how you structure it, this actually has to be either:

```
struct Element { bot_name: Option<HeaderValueString>, rules: Vec<Rule> }
```

or

```
enum Element {
    BotName(HeaderValueString),
    Rule(Rule),
}
```

Because when a botname is mentioned it applies to all rules that follow it, until another botname is mentioned